rules/aws_cloudtrail_rules/aws_console_login.py

```diff
@@ -1,2 +1,10 @@
 def rule(event):
     return event.get("eventName") == "ConsoleLogin"
+
+
+def alert_context(event):
+    context = {}
+    context["ip_and_username"] = event.get(
+        "sourceIPAddress", "<MISSING_SOURCE_IP>"
+    ) + event.deep_get("userIdentity", "userName", default="<MISSING_USER_NAME>")
```

Comment:
Cloudtrail logs don't have a `userIdentity.userName` field.  You'll need to parse the name from after the last `/` in `userIdentity.arn`

arn:aws:sts::98765432:assumed-role/IAM/**alice**